File: java/spring-session/web_redis_hook_repository/src/main/java/sample/controller/SampleController.java
package sample.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpSession;
import java.io.Serializable;
import java.util.Optional;

@RestController
public class SampleController {
    private static final String DATA = "data";

    @Autowired
    private HttpSession session;

    @GetMapping("/set/{value}")
    public Data set(@PathVariable("value") String value) {
        var data = new Data(value);
        session.setAttribute(DATA, data);

        return data;
    }

    @GetMapping("/get")
    public Optional<Data> get() {
        var data = (Data)session.getAttribute(DATA);
        session.invalidate();

        return Optional.ofNullable(data);
    }

    public record Data(String value) implements Serializable {}
}

File: kotlin/springfox/sample2/src/main/kotlin/sample/SampleController.kt
package sample

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class SampleController {
    @GetMapping("/")
    fun home() = "home page"

    @GetMapping("/data/{id}")
    fun data(@PathVariable("id") id: String) = "data-${id}"
}